testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -1233,6 +1235,17 @@ protected static SortedSet<String> keywords(@Nullable String dialect) {
         .fails("Bang equal '!=' is not allowed under the current SQL conformance level");
   }
 
+  @Test void testRecursiveQueryCloned() throws Exception {
```

Comment:
May be useful to add a Javadoc comment here with the issue that is being solved.

Reply:
Thanks @mihaibudiu for the review, added a Javadoc comment to the test case.